### PR TITLE
fix(ci): fix there is no reactor running in meta recovery tests, when kill-meta enabled

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Apply patch: https://github.com/madsim-rs/madsim/pull/231 which fixes an upstream bug: https://github.com/madsim-rs/madsim/issues/228.

This patch fixes madsim-sqlx's drop on spawn error, and fixes one of the bugs in the meta recovery tests.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -354,6 +354,7 @@ tokio-postgres = { git = "https://github.com/madsim-rs/rust-postgres.git", rev =
 # If we can merge in: https://github.com/madsim-rs/sqlx/pull/2,
 # we can change it to patch madsim version instead.
 sqlx = { git = "https://github.com/kwannoel/sqlx.git", rev = "ddf222f56cf99f865231a5383053645c6ea05ba3" }
+madsim = { git = "https://github.com/madsim-rs/madsim.git", rev = "f0ffd24864245aea236833f1ff3638f75bd51700" }
```
Comment (kwannoel) [inline Cargo.toml:357]: We can remove once https://github.com/madsim-rs/madsim/pull/231 merged.

File: ci/scripts/deterministic-recovery-test.sh [hunk 2]
```diff
@@ -44,7 +44,7 @@ filter_stack_trace_for_all_logs() {
   done
 }
 
-trap filter_stack_trace_for_all_logs ERR
+# trap filter_stack_trace_for_all_logs ERR
```
Comment (kwannoel) [inline ci/scripts/deterministic-recovery-test.sh:47]: We can just remove the log filter, until recovery tests stabilize. The recent logs are < 1MB, and we only upload the zipped archive for the logs.

**Comment by kwannoel** [comment]
The original bug of `reactor missing` is no longer present. Instead we have other bugs. I have included them in this branch: https://github.com/risingwavelabs/risingwave/pull/18855.

I will just disable kill-meta in this branch, then we can merge it in. Disabling kill-meta is required since there are still bugs that occur semi-frequently, so we don't want to block other's PR workflows.

File: Cargo.toml (hunk 1 above)
Comment (kwannoel) [reply Cargo.toml:357]: https://github.com/risingwavelabs/risingwave/issues/18856

**Comment by BugenZhao** [comment]
> The original bug of `reactor missing` is no longer present.

Sounds good!

**Review by BugenZhao — APPROVED** [review]

**Review by yezizp2012 — APPROVED** [review]
LGTM!

**Pull request merged**